src/config.rs

```diff
@@ -18,6 +19,7 @@ pub struct Config {
     package_dir: PathBuf,
     cache_dir: PathBuf,
     links: Vec<Link>,
+    #[serde(default)]
```

Comment:
Wouldn't an `Option` be more appropriate than an empty `String` to indicate the absence of token ?

Reply:
Hmm I don't get it. There's no empty string, nor empty token.

If the token is empty then a hard-error is returned and the Config isn't created, so replacing it by an Option would change nothing: I would still error on None, and still only accept non-empty String as a Token for it to pass the constructor's restrictions.

If I misunderstood you, please elaborate.

Reply:
Nevermind, actually in the changes introduced by this commit you set the value of the string in the structure, then check for emptiness before raising an error.
However, this is only done internally in `load_from` (which is not expanded fully in the diff and thus was confusing).